src/components/drawing/index.js

```diff
@@ -192,6 +192,55 @@ function setFillStyle(sel, trace, gd) {
             undefined, markerPattern.fillmode,
             patternBGColor, patternFGColor, patternFGOpacity
         );
+    } else if(trace.fillgradient && trace.fillgradient.orientation !== 'none') {
```

Comment:
`trace.fillgradient` is used in several places below.
How about adding
```js
var fillgradient = trace.fillgradient;
```
to the top and then use it from the variable not the trace?
It should help reduce the bundle size, may run faster and avoid potential typos.